node/src/components/storage/lmdb_ext.rs

```diff
@@ -211,12 +240,54 @@ pub(super) fn deserialize<T: DeserializeOwned>(raw: &[u8]) -> Result<T, LmdbExtE
     bincode::deserialize(raw).map_err(|err| LmdbExtError::DataCorrupted(Box::new(err)))
 }
 
+/// Returns `true` if the specified bytes represent the legacy version of `UnbondingPurse`.
+fn is_legacy(raw: &[u8]) -> bool {
+    !raw.starts_with(UNBONDING_PURSE_V2_MAGIC_BYTES)
+}
+
+/// Deserializes `UnbondingPurse` from a buffer.
+pub(super) fn deserialize_unbonding_purse<T: DeserializeOwned>(
+    raw: &[u8],
+) -> Result<T, LmdbExtError> {
+    if is_legacy(raw) {
+        #[derive(Deserialize)]
+        pub struct LegacyUnbondingPurse {
+            bonding_purse: URef,
+            validator_public_key: PublicKey,
+            unbonder_public_key: PublicKey,
+            era_of_creation: EraId,
+            amount: U512,
+        }
+        let deserialized: LegacyUnbondingPurse = bincode::deserialize(raw).unwrap();
+        let upgraded_unbonding_purse = UnbondingPurse::new(
+            deserialized.bonding_purse,
+            deserialized.validator_public_key,
+            deserialized.unbonder_public_key,
+            deserialized.era_of_creation,
+            deserialized.amount,
+            Option::default(),
+        );
+        let raw = bincode::serialize(&upgraded_unbonding_purse).unwrap();
+        bincode::deserialize(&raw).map_err(|err| LmdbExtError::DataCorrupted(Box::new(err)))
```

Comment:
Note to self:
This roundtrip should go away.

Reply:
It's gone thanks to https://github.com/casper-network/casper-node/pull/2967#discussion_r866926169